freqtrade/exchange/binance.py

```diff
@@ -89,3 +96,109 @@ def stoploss(self, pair: str, amount: float, stop_price: float, order_types: Dic
                 f'Could not place sell order due to {e.__class__.__name__}. Message: {e}') from e
         except ccxt.BaseError as e:
             raise OperationalException(e) from e
+
+    def transfer(self, asset: str, amount: float, frm: str, to: str, pair: Optional[str]):
+        res = self._api.sapi_post_margin_isolated_transfer({
+            "asset": asset,
+            "amount": amount,
+            "transFrom": frm,
+            "transTo": to,
+            "symbol": pair
+        })
+        logger.info(f"Transfer response: {res}")
```

Comment:
As mentioned before - please focus on cross-margin - or remove support for binance for now completely.

I see isolated margin out of scope for the moment (as mentioned before already) - as it includes/introduces another level of complexity by having to transfer to the isolated margin account first.

That can be an enhancement added once stability of the whole short-logic is confirmed and merged - but not until then - and i'll reject all Pull requests that add parts of this.

Reply:
If simplicity is what you're going for, I don't think cross-margin is the way to do it, I created another PR #5339 which adds skeleton functions for implementing cross margin, the most complicated part would be keeping an accurate calculation of the maintenance margin, the 1m candles wouldn't be good enough because one candle could be at  it's peak while another could be at it's low. Is there another way that freqtrade keeps track of the price? Is the price calculated in realtime for the stoploss?

I'll remove this transfer stuff from this PR. 

I think the least complicated would be isolated futures, there's no transfer involved, I can look at FTX also

Reply:
You will need the logic for cross-margin for kraken anyway.

the whole margin handling must be "mostly" transparent to the bot anyway - so "most" of the logic (if it's 1 call to borrow, "automatic borrow" like in kraken, or 20 calls to borrow) must all be "hidden" inside the exchange class. 
Now you'll need one call to "repay" eventually (that does nothing in the regular case) - but this should be the exception, not the rule.

Otherwise you'll end up with a lot of "if exchange = kraken then this" logic - which i'd only really want in very limited sections where it can't be avoided.